cpp/example/simple_kv_store.cc

```diff
@@ -0,0 +1,210 @@
+
+ private:
+  // Get all data from MainServer.
+  void HanldeFailover();
+
+  std::unordered_map<std::string, std::string> data_;
+  std::mutex mtx_;
+};
+
+BackupServer::BackupServer() {
+  // Handle failover when the actor restarted.
+  if (ray::WasCurrentActorRestarted()) {
+    HanldeFailover();
+  }
+  RAYLOG(INFO) << "BackupServer created";
+}
+
+std::unordered_map<std::string, std::string> BackupServer::GetAllData() {
+  std::unique_lock<std::mutex> lock(mtx_);
+  return data_;
+}
+
+void BackupServer::SyncData(const std::string &key, const std::string &val) {
+  std::unique_lock<std::mutex> lock(mtx_);
+  data_[key] = val;
+}
+
+class MainServer {
+ public:
+  MainServer();
+
+  void Put(const std::string &key, const std::string &val);
+
+  std::pair<bool, std::string> Get(const std::string &key);
+
+  std::unordered_map<std::string, std::string> GetAllData();
+
+ private:
+  void HanldeFailover();
+
+  std::unordered_map<std::string, std::string> data_;
+  std::mutex mtx_;
+
+  ray::ActorHandle<BackupServer> backup_actor_;
+};
+
+MainServer::MainServer() {
+  if (ray::WasCurrentActorRestarted()) {
+    HanldeFailover();
+  }
+
+  backup_actor_ = *ray::GetActor<BackupServer>(BACKUP_SERVER_NAME);
+  RAYLOG(INFO) << "MainServer created";
+}
+
+void MainServer::Put(const std::string &key, const std::string &val) {
+  // SyncData before put data.
+  auto r = backup_actor_.Task(&BackupServer::SyncData).Remote(key, val);
+  std::vector<ray::ObjectRef<void>> objects{r};
+  auto result = ray::Wait(objects, 1, 2000);
+  if (result.ready.empty()) {
+    RAYLOG(WARNING) << "MainServer SyncData failed.";
+  }
+
+  std::unique_lock<std::mutex> lock(mtx_);
+  data_[key] = val;
+}
+
+std::pair<bool, std::string> MainServer::Get(const std::string &key) {
+  std::unique_lock<std::mutex> lock(mtx_);
+  return common::Get(key, data_);
+}
+
+std::unordered_map<std::string, std::string> MainServer::GetAllData() {
+  std::unique_lock<std::mutex> lock(mtx_);
+  return data_;
+}
+
+void BackupServer::HanldeFailover() {
+  // Get all data from MainServer.
+  auto dest_actor = *ray::GetActor<MainServer>(MAIN_SERVER_NAME);
+  data_ = *dest_actor.Task(&MainServer::GetAllData).Remote().Get();
+  RAYLOG(INFO) << "BackupServer get all data from MainServer";
+}
+
+void MainServer::HanldeFailover() {
+  // Get all data from BackupServer.
+  backup_actor_ = *ray::GetActor<BackupServer>(BACKUP_SERVER_NAME);
+  data_ = *backup_actor_.Task(&BackupServer::GetAllData).Remote().Get();
+  RAYLOG(INFO) << "MainServer get all data from BackupServer";
+}
+
+static MainServer *CreateMainServer() { return new MainServer(); }
+
+static BackupServer *CreateBackupServer() { return new BackupServer(); }
+
+RAY_REMOTE(CreateMainServer, &MainServer::GetAllData, &MainServer::Get, &MainServer::Put);
+
+RAY_REMOTE(CreateBackupServer, &BackupServer::GetAllData, &BackupServer::SyncData);
+
+ray::PlacementGroup CreateSimplePlacementGroup(const std::string &name) {
```

Comment:
The name `CreateSimplePlacementGroup` is confusing. I'd suggest just moving this to the `StartServer` function